cpp/farm_ng/core/prototools/event_log_reader.cpp

```diff
@@ -20,54 +20,67 @@ namespace farm_ng {
 class EventLogReaderBinaryImpl : public EventLogReaderImpl {
  public:
   EventLogReaderBinaryImpl(std::filesystem::path const& log_path)
-      : log_path(log_path), in(log_path.string(), std::ofstream::binary) {
+      : log_path(log_path) {
+    reset();
+  }
+
+  virtual void reset() {
+    in = std::ifstream(log_path.string(), std::ofstream::binary);
     if (!in) {
-      throw std::runtime_error(
-          FARM_FORMAT("Could not open file: {}", log_path));
+      throw EventLogExist(FARM_FORMAT("Could not open file: {}", log_path));
     }
   }
-
   std::string readBytes(uint64_t n_bytes) {
+    FARM_CHECK(in.good());
```

Comment:
Is this violation a precondition or class invariant. Or should we throw here?

Reply:
This a precondition. As the exception should have been handled before this point. Also if an exception is thrown, the io stream is reset.